### PR TITLE
feat(demo): adds OTHER_JS_FILE option and exposes hookstore and config store

### PR DESCRIPTION
**Pull request by scefali**
This PR does the following:
1. Adds `OTHER_JS_FILE` option which will get added to the HTML of the main app
2. Exposes the `HookStore` on `ConfigStore` on the global window

The purpose of this PR is so that for the Sentry Demo, we can add Amplitude analytics by creating a JS file (`OTHER_JS_FILE`) and adding callbacks to the analytics hooks. Related PR: https://github.com/getsentry/demo-add-on/pull/1

### CODE REVIEW

File: src/sentry/templates/sentry/layout.html [hunk 1]
```diff
@@ -49,6 +49,11 @@
   {% asset_url "sentry" "dist/app.js" as asset_url %}
   {% script src=asset_url data-entry="true" %}{% endscript %}
 
+  {% other_js_file as asset_url %}
+  {% if asset_url %}
+    {% script src=asset_url %}{% endscript %}
+  {% endif %}
```
Comment (scefali) [inline src/sentry/templates/sentry/layout.html:55]: @markstory Is this the right way to add an optional script tag? Note this file is coming from the CDN and not Sentry.

File: src/sentry/conf/server.py [hunk 2]
```diff
@@ -2119,4 +2119,8 @@ def create_partitioned_queues(name):
 # Enabling this will allow users to create accounts without an email or password.
 DEMO_MODE = False
 
+# all demo orgs are owned by the user with this email
 DEMO_ORG_OWNER_EMAIL = None
+
+# adds an extra JS to HTML template
+OTHER_JS_FILE = None
```
Comment (EvanPurkhiser) [inline src/sentry/conf/server.py:2126]: Can you make this accept an array and maybe call it like

```
INJECTED_SCRIPT_ASSETS
```

File: src/sentry/static/sentry/app/stores/configStore.tsx [hunk 3]
```diff
@@ -101,4 +101,6 @@ type ConfigStore = Reflux.Store & ConfigStoreInterface;
 
 const ConfigStore = Reflux.createStore(configStoreConfig) as ConfigStore;
 
+window.ConfigStore = ConfigStore;
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/stores/configStore.tsx:104]: Don't do this here, we do this already https://github.com/getsentry/sentry/blob/3f9c71202f641949a120528db4bc60a6ffc9838a/src/sentry/static/sentry/app/bootstrap.tsx#L199

File: src/sentry/static/sentry/app/stores/hookStore.tsx [hunk 4]
```diff
@@ -120,4 +120,6 @@ type HookStore = Reflux.Store & HookStoreInterface;
  */
 const HookStore = Reflux.createStore(hookStoreConfig) as HookStore;
 
+window.HookStore = HookStore;
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/stores/hookStore.tsx:123]: Let's do this in bootstrap.tsx

https://github.com/getsentry/sentry/blob/3f9c71202f641949a120528db4bc60a6ffc9838a/src/sentry/static/sentry/app/bootstrap.tsx#L184-L203

File: src/sentry/templates/sentry/layout.html (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/templates/sentry/layout.html:55]: Any chance you could make it to multiple?

Comment (scefali) [reply src/sentry/templates/sentry/layout.html:55]: @EvanPurkhiser make what to multiple?

Comment (EvanPurkhiser) [reply src/sentry/templates/sentry/layout.html:55]: Like can we have multiple additional assets, that's why I suggested `INJECTED_SCRIPT_ASSETS` (plural), so you could specify a list of files

Comment (scefali) [reply src/sentry/templates/sentry/layout.html:55]: @EvanPurkhiser got it, i'll give it a shot

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**